controllers/argocd/role.go

```diff
@@ -279,59 +279,88 @@ func (r *ReconcileArgoCD) reconcileRoleForApplicationSourceNamespaces(name strin
 	return nil
 }
 
-func (r *ReconcileArgoCD) reconcileClusterRole(name string, policyRules []v1.PolicyRule, cr *argoproj.ArgoCD) (*v1.ClusterRole, error) {
+func (r *ReconcileArgoCD) reconcileClusterRole(componentName string, policyRules []v1.PolicyRule, cr *argoproj.ArgoCD) (*v1.ClusterRole, error) {
 
 	allowed := false
 	if allowedNamespace(cr.Namespace, os.Getenv("ARGOCD_CLUSTER_CONFIG_NAMESPACES")) {
+		// namespace is allowed to host cluster-scoped Argo CD instance
 		allowed = true
 	}
 
-	// Check if it is cluster-scoped instance namespace and user doesn't want to use default ClusterRole
-	if allowed && cr.Spec.DefaultClusterScopedRoleDisabled {
-
-		// In case DefaultClusterScopedRoleDisabled was false earlier and default ClusterRole was created, then delete it.
-		existingClusterRole := &v1.ClusterRole{}
-		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: GenerateUniqueResourceName(name, cr)}, existingClusterRole); err == nil {
+	if allowed && cr.Spec.DefaultClusterScopedRoleDisabled && cr.Spec.AggregatedClusterRolesEnabled {
+		return nil, fmt.Errorf("Custom Cluster Roles and Aggregated Cluster Roles can not be used together.")
```

Comment:
For this one, I was thinking something like this, here:

```go
	if allowed && cr.Spec.DefaultClusterScopedRoleDisabled && cr.Spec.AggregatedClusterRolesEnabled {
		log.Error(fmt.Errorf("Unable to create ClusterRole: custom Cluster Roles and Aggregated Cluster Roles can not be used together in '%s' of '%s. Update ArgoCD CR to ensure that either .spec.defaultClusterScopedRoleDisabled or .spec.aggregatedClusterRoleEnabled is true, but NOT both at the same time.'", cr.Name, cr.Namespace), "")
		return nil, nil
	}
```

Unfortunately at the moment we don't have `.status.conditions` in ArgoCD CR, but I agree they would 100% be helpful.  The best we can do in this case is just log errors to the console, and expect the user to check there :shrug:. 

Now, since we are not returning an error, what this means is that the rest of the code will continue. So a  `ClusterRoleBinding` will be created, and point to a `ClusterRole` that doesn't exist, but I think this is fine (not perfect, but fine, Kubernetes is OK when this happenes).